### PR TITLE
Documentation: Add instructions for installing cilium on Rancher Desktop

### LINKED ISSUE #18675: Can't start cilium pod after installation on rancher-desktop v1.0.0
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

Wanted to created a playground with k3s, cilium, prometheus/grafana and used rancher-desktop on Linux (ubuntu 20.04 based).
Used this guide: 
[https://docs.cilium.io/en/v1.11/gettingstarted/grafana/](url)
After installation via helm, cilium pod report an error:
`Error: failed to generate container "5ac671b188b972f82f131d21e54274b13d6aeaa46413ae8dd44db4e872fbe182" spec: failed to generate spec: path "/run/cilium/cgroupv2" is mounted on "/run/cilium/cgroupv2" but it is not a shared or slave mount`

```
kubectl get pods -A
NAMESPACE     NAME                                      READY   STATUS                      RESTARTS   AGE
kube-system   local-path-provisioner-84bb864455-nvpwg   1/1     Running                     0          23m
kube-system   helm-install-traefik-crd--1-fkzvz         0/1     Completed                   0          23m
kube-system   metrics-server-ff9dbcb6c-mzqcn            1/1     Running                     0          23m
kube-system   svclb-traefik-hf2q2                       2/2     Running                     0          22m
kube-system   helm-install-traefik--1-kkrzb             0/1     Completed                   2          23m
kube-system   traefik-55fdc6d984-lgqxk                  1/1     Running                     0          22m
kube-system   cilium-operator-5ffd7d9795-m8w2v          0/1     Pending                     0          6m7s
kube-system   cilium-operator-5ffd7d9795-z9ts6          1/1     Running                     0          6m7s
kube-system   coredns-96cc4f57d-2zl62                   1/1     Running                     0          23s
kube-system   cilium-rqzk4                              0/1     Init:CreateContainerError   0          6m7s
```

I got suggestion on forums to use workaround to mount manually:
`mount bpffs /sys/fs/bpf -t bpf`

but the question is where to mount? I am just a beginner in k8s, so a bit lost there. Rancher-desktop spins off a VM via qemu and entire K3S cluster runs inside that VM, so does the cilium as well. I can ssh to VM, but can't mount anything in it, It has no effect. Can this WA be applied somewhere else in k3s cluster? 

### Cilium Version

v1.11.1

### Kernel Version

Linux lima-rancher-desktop 5.10.88-0-virt #1-Alpine SMP Tue, 28 Dec 2021 12:51:14 +0000 x86_64 Linux

(This is kernel of the VM that runs k3s)

### Kubernetes Version

Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.3", GitCommit:"816c97ab8cff8a1c72eccca1026f7820e93e0d25", GitTreeState:"clean", BuildDate:"2022-01-26T02:20:15Z", GoVersion:"go1.17.6", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.6+k3s1", GitCommit:"3228d9cb9a4727d48f60de4f1ab472f7c50df904", GitTreeState:"clean", BuildDate:"2022-01-25T01:27:44Z", GoVersion:"go1.16.10", Compiler:"gc", Platform:"linux/amd64"}

### Sysdump

[cilium-sysdump-20220131-152742.zip](https://github.com/cilium/cilium/files/7971475/cilium-sysdump-20220131-152742.zip)

### Relevant log output

_No response_

### Anything else?

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by chancez**
Signed-off-by: Chance Zibolski <chance.zibolski@gmail.com>

Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
ium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Thanks for contributing!

Fixes: #18675

```release-note
Document installing Cilium on Rancher Desktop
```

Depends on https://github.com/cilium/cilium-cli/pull/744 to correctly auto-detect CNI location on Rancher Desktop.

### CODE REVIEW

File: Documentation/gettingstarted/k8s-install-default.rst [hunk 1]
```diff
@@ -199,6 +199,16 @@ to create a Kubernetes cluster locally or using a managed Kubernetes service:
           ``--cni=cilium`` parameter in ``minikube start`` command. However, this may not
           install the latest version of cilium.
 
+    .. group-tab:: Rancher Desktop
+
+       Install Rancher Desktop >= v1.1.0 as per Rancher Desktop documentation:
```
Comment (sayboras) [inline Documentation/gettingstarted/k8s-install-default.rst:204]: I was having 1.0.1 locally, just curious why this version is not working.

**Review by sayboras — APPROVED** [review]
This is awesome 💯 

I managed to follow the docs, and have cilium running with rancher-desktop on m1 mac. Connectivity check is passed as well.

<details>

```
🏃 Running tests...

[=] Test [no-policies]
....................
[=] Test [allow-all]
................
[=] Test [client-ingress]
..
[=] Test [echo-ingress]
..
[=] Test [client-egress]
..
[=] Test [to-entities-world]
......
[=] Test [to-cidr-1111]
....
[=] Test [echo-ingress-l7]
..
[=] Test [client-egress-l7]
........
[=] Test [dns-only]
........
[=] Test [to-fqdns]
......
✅ All 11 tests (76 actions) successful, 0 tests skipped, 0 scenarios skipped.
```

</details>

File: Documentation/gettingstarted/rancher-desktop.rst [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    https://docs.cilium.io
+
+.. _rancher_desktop_install:
+
+*************************************
+Getting Started Using Rancher Desktop
+*************************************
+
+This guide walks you through installation of Cilium on `Rancher Desktop <https://rancherdesktop.io>`_,
+An open-source desktop application for Mac, Windows and Linux.
```
Comment (qmonnet) [inline Documentation/gettingstarted/rancher-desktop.rst]:
```suggestion
an open-source desktop application for Mac, Windows and Linux.
```

File: Documentation/gettingstarted/k8s-install-helm.rst [hunk 3]
```diff
@@ -227,6 +227,22 @@ Install Cilium
           helm install cilium |CHART_RELEASE| \\
              --namespace $CILIUM_NAMESPACE
 
+    .. group-tab:: Rancher Desktop
+
+       To install Cilium on `Rancher Desktop <https://rancherdesktop.io>`_,
+       perform the following steps:
+
+       .. include:: rancher-desktop-configure.rst
```
Comment (qmonnet) [inline Documentation/gettingstarted/k8s-install-helm.rst:237]:
```suggestion
       **Configure Rancher Desktop**

       .. include:: rancher-desktop-configure.rst
```

**Review by qmonnet — CHANGES_REQUESTED** [review]
Looks good, thanks! Please find two nits below.

File: Documentation/gettingstarted/k8s-install-default.rst (hunk 1 above)
Comment (chancez) [reply Documentation/gettingstarted/k8s-install-default.rst:204]: You need 1.1 for https://github.com/rancher-sandbox/rancher-desktop/pull/1565, without this setting more than one flag in `K3S_EXEC` fails.

**Review by qmonnet — APPROVED** [review]

File: Documentation/gettingstarted/k8s-install-default.rst (hunk 1 above)
Comment (ctmach) [reply Documentation/gettingstarted/k8s-install-default.rst:204]: oh thanks, no wonder i used to have issue invalid flag --disable...

Comment (sayboras) [reply Documentation/gettingstarted/k8s-install-default.rst:204]: oh thanks, no wonder I was having weird issue invalid flag (e.g. --disable-network-policy, etc)

**Pull request merged**